docs/adr/0018-use-tailwind-with-custom-scss.md

```diff
@@ -0,0 +1,136 @@
+# Use Utility-First CSS (Tailwind) with Custom SCSS
+
+- Status: proposed
+- Deciders: Jody Heavener, Lauren Zugai, Danny Coates
+- Date: 2020-05-08
+
+## Context and Problem Statement
+
+The [Settings Redesign project](https://github.com/mozilla/fxa/issues/3740) provides us with an opportunity to review how FxA approaches and employs CSS, both while building out new components for this project and for FxA going forward.
+
+Historically, the Firefox Accounts codebase has not adhered to a formal CSS structure. This ADR serves to determine how we'll approach our CSS architecture in the Settings Redesign project, evaluating libraries and frameworks to determine which if any will be the best option for the FxA ecosystem. It is part 2 of two [Settings Redesign CSS ADRs](https://github.com/mozilla/fxa/issues/5087); part 1, detailing how we'll approach build conventions and variables, [can be found here](https://github.com/mozilla/fxa/blob/master/docs/adr/0015-use-css-variables-and-scss.md).
+
+Considerations around class naming conventions, color and measurement standards, interoperability across shared components, and custom configuration options offered by each library and framework to meet Settings Redesign design standards are taken into account. Noteably, the new design uses space measurements in increments of 8px and [colors](https://protocol.mozilla.org/fundamentals/color.html) are based in Mozilla Protocol's design system, where a hue's brightness scales in increments of 10.
+
+## Decision Drivers
+
+- **Reusability** - does the approach yield DRY, lean code that can be reused and repurposed?
+- **Longevity** - will the approach be supported in upcoming years and will it provide a stable platform for years of revolving HTML through added features and bug fixes?
+- **Developer experience** - are some team members already familiar with the approach, making the transition easier than an unfamiliar one?
+- **Ease of use** - will the choice result in a large learning curve or be easy for engineers to pick up?
+
+## Considered Options
+
+- Option A - Use an existing UI library or framework
+- Option B - Use an existing utility library
+- Option C - Develop our own hybrid component and utility framework
+- Option D - Custom SCSS with loose structure
+
+## Decision Outcome
+
+Chosen options: "Option B" with Tailwind CSS for majority styling, and implementation details from "Option D" when utility classes don't meet the entire need, because:
```

Comment:
Would we actively discourage the use of `Option D`? I've found it useful for `!important` that the developer documents exactly why they had to introduce this and maybe things they tried. We could do something similar for this.

Reply:
I think Option D (needing to use custom styles) will inevitably bleed into any of the chosen solutions at some point. While not true 100% of the time, !important often hints at a larger issue, so I definitely like the idea of explaining why you absolutely need to use it. Would it help if we outlined that and some additional "free-styling" guidance in the decision outcome?

Reply:
I think it should definitely be discouraged _if we have a Tailwind class that would apply the same style_. Some things are inevitable and don't need an explanation, like background-images set on devices & apps. I think we'll have a lot of elements that can be properly styled with only utility classes so I like the idea of a comment at the top of a set of custom styles. 🙂 If anything, it'll serve as a reminder for someone else going in to edit that file that there's Tailwind classes available and you shouldn't just go hacking away at the CSS. Jody, we could add this example when one of us grabs #5244 next sprint.

(I also _really_ dislike `!important`. 😅)